### ISSUE TITLE
Tmux cursor position shifts

### ISSUE THREAD

**Issue by ddkn** [martanne/vis#763]
Hello,

I have been having an issue with moving the cursor (via movement keys) which seem to shift the previous text over one, such that all of the text becomes staggered, or the text shifts as I move. Has anyone run into this issue? It becomes unsightly when writing latex text as you lose track of the actual cursors position. I have row numbers turned off, and minimial settings in `~/.tmux.conf` and `~/.config/vis/visrc.lua`.

The problem does not happen if I choose to run vis without invoking tmux. I am running on OpenBSD 6.5 stable using xterm. Here are my config files:

```
# ~/.tmux.conf
set -g default-terminal "tmux-256color"
set -g escape-time 10
set-option -g status-position top
set-option -g mouse on
```

```
-- ~/.config/vis/visrc.lua
require('vis')

function latex_style()
        vis:command('set tw 2')
        vis:command('set et off')
        vis:command('set cc 0')
        vis:info('LaTeX')
end

function python_style()
        vis:command('set tw 4')
        vis:command('set et on')
        vis:command('set cc 79')
        vis:info('Python PEP8')
end

function default_style()
        vis:command('set tw 4')
        vis:command('set et off')
        vis:command('set cc 79')
        vis:info('Default')
end

function trimspace()
        vis:command('x/[ \\t\\r]+$/c//')
        vis:info('Trim whitespace')
end

vis.events.subscribe(vis.events.INIT, function()
        -- Your global configuration options
end)

vis.events.subscribe(vis.events.WIN_OPEN, function(win)
        -- Your per window configuration options e.g.
        -- vis:command('set rnu on')
        vis:command('set tw 4')
        vis:command('set cc 79')
        vis:command('set ai on')
        vis:command('set show-tabs on')
        vis:command('set syntax off')
        if win.syntax == 'python' then
                python_style()
        elseif win.syntax == 'latex' then
                latex_style()
        end
end)

vis:map(vis.modes.NORMAL, "<F5>", trimspace)
vis:map(vis.modes.NORMAL, "<F6>", default_style)
vis:map(vis.modes.NORMAL, "<F7>", python_style)
vis:map(vis.modes.NORMAL, "<F8>", latex_style)
```

Any advice would be greatly appreciated! Until then I am either not using tmux, or using vi inside of tmux.

Thanks!

**Comment by ddkn** [comment]
So after some playing around one of the major issues seems to be the ``vis:command('set show-tabs on')`` option. Setting ``set show-tabs off`` removes the issue and is more manageable.

Also some of the colour of cursor issues was mitigated with the removal of ``set -g default-terminal "tmux-256color"``. Perhaps this is due to not having the default xterm behaviour as 256 colour.

**Comment by ddkn** [comment]
This issue was not present on MacOS High Sierra with tmux. I suspect the aliased/staggering text with `set show-tabs on` is due to either font or locale on OpenBSD. 

![stagger](https://user-images.githubusercontent.com/13819239/82104115-f62f5700-96eb-11ea-9254-7fcbb873d880.png)

You can see as I move the cursor near the tab character it doesn't refresh the screen as expected in tmux. Also, the colorcolumn `set cc 79` has some issues as well. It works just fine outside of tmux on OpenBSD.

On OpenBSD I was using Fira Sans in xterm. I switiched it to DeJaVu Mono with no luck. The locales were set as such:

```
export LC_ALL=en_US.UTF-8
export LC_CTYPES=en_US.UTF-8
```

I am a bit stumped currently.

**Comment by erf** [comment]
I have experienced similar when there are very weird unicode characters on screen, see #524 

but it seem like an issue with tmux. did you search issues with `tmux`?

did you build `vis` using `ncursesw` (wide characters) ?

**Comment by ddkn** [comment]
I recently upgraded to OpenBSD 6.7. I set `LC_ALL=en_US.UTF-8` which now worked, but the wrong UTF-8 characters were displaying for `set show-tabs on`. After running `tmux -u` for forcing utf-8 support, vis worked as expected!

Alternatively I set in my `.kshrc` (according to tmux's man page):
~~~
LC_ALL=en_US.UTF-8
LC_CTYPE=en_US.UTF-8
LANG=en_US.UTF-8
~~~
and can run `tmux` (without `-u`). Now causes `vis/vise` to work correctly.

Thanks @erf for pointing out to look at tmux for utf support! I will be closing this issue as it is now solved.
